processor/processor.go

```diff
@@ -506,6 +476,14 @@ WORKERPOOL_LOOP:
 
 				// backoff & wait for workers to finish or a job to be queued
 				time.Sleep(backoffDuration)
+
+				if updateTimeoutCnt == 0 {
+					// I am still alive, update expire
+					var newTimeout int64 = time.Now().Unix() + job.AggregationTimeout
+					wp.p.Storage.Redis.HSet(storage.AggrKeyPrefix+wp.aggr.ID, "ExpiresAt", newTimeout)
+				}
+				updateTimeoutCnt = (updateTimeoutCnt + 1) % 20
```

Comment:
Could you please elaborate a bit on the _updateTimeoutCnt_ usage?

Reply:
This aims to solve the issue that an aggregation might be processed for longer than the default timeout. We extend (by setting again) the timeout. But in order to avoid making a lot of calls to redis, we do this frequently, currently every 20 iterations.

Reply:
Ok, thank you for clarifying this. So the modulo 20 means that the **ExpiresAt** field will be updated every 20 jobs right? Let's add a comment for that behavior, it would heavily facilitate a future contributor IMHO.